### PR TITLE
test expected output

### PR DESCRIPTION
**Pull request by Bchass**
## Description
Fixes https://github.com/vprusso/toqito/issues/471

## Changes

- update examples to try and line up with what doctest expects

## Checklist

  -  [x] Use `ruff` and `pylint` for errors related to code style and formatting.
  -  [ ] Verify all previous and newly added unit tests pass in `pytest`.
  -  [ ] Check the documentation build does not lead to any failures. `Sphinx` build can be checked locally for any failures related to your PR
  -  [x] Use `linkcheck` to check for broken links in the documentation
  -  [x] Use `doctest` to verify the examples in the function docstrings work as expected.

### CODE REVIEW

**Comment by purva-thakre** [comment]
You might want to rebase your PR. What you're working with is from an older commit.

Also, you might have to make changes to the code of the function to get to `array` output (probably)

image

**Comment by purva-thakre** [comment]
@Bchass You could try to figure out where the function changes a numpy array to a 2D list at each step by making python print the variable type. 

https://github.com/vprusso/toqito/blob/eb5820721884dae5b9fdaad68913b2d96785167d/toqito/matrices/cyclic_permutation.py#L57-L62

For example, temporarily use `print(type(variable_name))` after each line in the function code. If you use the function, you should be able see where the function transitions from manipulating a `np.ndarray` type to a `2D list`. 

Hopefully, this makes sense? Again, I haven't had the chance to dig into the issue yet but this is what my first step would have been.

**Comment by Bchass** [comment]
Thanks for the info! I'll look into it more at some point this weekend.

**Comment by Bchass** [comment]
@purva-thakre Stepping through the function, it never converted to a 2D list. It boiled down to whitespace and how I was calling the function in the examples.

Came across: https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html and was able to format the result how `doctest` wanted it.

**Comment by purva-thakre** [comment]
>Stepping through the function, it never converted to a 2D list. It boiled down to whitespace and how I was calling the function in the examples.

That's interesting!

File: toqito/matrices/cyclic_permutation.py [hunk 1]
```diff
@@ -59,4 +56,5 @@ def cyclic_permutation_matrix(n: int, k: int = 1) -> np.ndarray:
     p_mat[0, -1] = 1
 
     result_mat = np.linalg.matrix_power(p_mat, k)
+    np.set_printoptions(edgeitems=n*n, linewidth=40)
```
Comment (purva-thakre) [inline toqito/matrices/cyclic_permutation.py:59]: Maybe you can get rid of this line. If the issue was related to using `print(cyclic_permutation_matrix(n, k))` vs `cyclic_permutation_matrix(n, k)`, then this addition is not required as you have corrected these lines in the example.

Comment (Bchass) [reply toqito/matrices/cyclic_permutation.py]: I'm sure I'll be taking on some more!

**Comment by Bchass** [comment]
More failures to investigate tomorrow

**Comment by purva-thakre** [comment]
@Bchass You can ignore the drop in `pytest` coverage. The `linkcheck` workflow passes now, previous failure could have been related to the being down being down or something.

**Review by purva-thakre — APPROVED** [review]

**Pull request merged**